panther_hardware_interfaces/test/panther_system/gpio/test_gpio_controller.cpp

```diff
@@ -13,36 +13,43 @@
 // limitations under the License.
 
 #include <cstdlib>
+#include <functional>
 
+#include <gmock/gmock.h>
 #include <gtest/gtest.h>
+
 #include <gpiod.hpp>
 
 #include <panther_hardware_interfaces/panther_system/gpio/gpio_controller.hpp>
 
 using GPIOInfo = panther_hardware_interfaces::GPIOInfo;
 using GPIOPin = panther_hardware_interfaces::GPIOPin;
 
-const std::vector<GPIOInfo> gpio_config_info_storage{
-  GPIOInfo{GPIOPin::WATCHDOG, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::AUX_PW_EN, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::CHRG_DISABLE, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::DRIVER_EN, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::E_STOP_RESET, gpiod::line::direction::INPUT},
-  GPIOInfo{GPIOPin::FAN_SW, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::GPOUT1, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::GPOUT2, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::GPIN1, gpiod::line::direction::INPUT},
-  GPIOInfo{GPIOPin::GPIN2, gpiod::line::direction::INPUT},
-  GPIOInfo{GPIOPin::SHDN_INIT, gpiod::line::direction::INPUT},
-  GPIOInfo{GPIOPin::VDIG_OFF, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::VMOT_ON, gpiod::line::direction::OUTPUT},
-  GPIOInfo{GPIOPin::CHRG_SENSE, gpiod::line::direction::INPUT},
-  GPIOInfo{GPIOPin::LED_SBC_SEL, gpiod::line::direction::OUTPUT},
+class MockGPIODriver : public panther_hardware_interfaces::GPIODriverInterface
+{
+public:
+  MOCK_METHOD(
+    void, GPIOMonitorEnable, (const bool use_rt, const unsigned gpio_monit_thread_sched_priority),
+    (override));
+  MOCK_METHOD(
+    void, ConfigureEdgeEventCallback, (const std::function<void(const GPIOInfo &)> & callback),
+    (override));
+  MOCK_METHOD(
+    void, ChangePinDirection, (const GPIOPin pin, const gpiod::line::direction direction),
+    (override));
+  MOCK_METHOD(bool, IsPinAvailable, (const GPIOPin pin), (const, override));
+  MOCK_METHOD(bool, IsPinActive, (const GPIOPin pin), (override));
+  MOCK_METHOD(bool, SetPinValue, (const GPIOPin pin, const bool value), (override));
 };
 
 class GPIOControllerWrapper : public panther_hardware_interfaces::GPIOControllerPTH12X
```

Comment:
We decide to not check GPIOControler for older version?

Reply:
I left it as it was (no unit tests) because of the following:
- we're approaching the end of new features for the older version
- the majority of gpio controller methods are not implemented for older version